packages/sanity/src/core/studio/components/navbar/presence/PresenceMenuItem.tsx

```diff
@@ -40,42 +36,47 @@ export const PresenceMenuItem = memo(function PresenceMenuItem(props: PresenceLi
   }, [menuItemElement, focused])
 
   const handleFocus = useCallback(() => {
-    onFocus(presence.user.id)
-  }, [onFocus, presence.user.id])
+    onFocus(user.id)
+  }, [onFocus, user.id])
 
-  const LinkComponent = useMemo(
-    () =>
-      // eslint-disable-next-line @typescript-eslint/no-shadow
-      forwardRef(function LinkComponent(linkProps, ref: ForwardedRef<HTMLAnchorElement>) {
-        if (!lastActiveLocation?.path) return null
-
-        return (
-          <IntentLink
-            {...linkProps}
-            intent="edit"
-            params={{
-              id: lastActiveLocation?.documentId,
-              path: PathUtils.toString(lastActiveLocation?.path),
-            }}
-            ref={ref}
-          />
-        )
-      }),
-    [lastActiveLocation],
```

Comment:
[For some reason React Compiler bails out of this component here](https://playground.react.dev/#N4Igzg9grgTgxgUxALhASwLYAcIwC4AEAVAQIZgEAKpeAFgKp5oA2FAZjBBgQOQACYUgDs0eAJ4B6KE2YSsNWmB4AdEdlyFguACYIYAITEBfAhy69mEbeVoq1OfAWDisCAgDFcAd1Ixd2gCUENgAaU29fQOCwjAQMCDCoMAQAYVJmZgAjUjgAa0TkgFE2NgQ4PAKEAFk4hIIkhABlPBoEEzNuHhgEHLw7TAdNAEkhPARRgBk0IVz2zk7BEXEJTmk9O1UBjScaoSghsYw58x4AOlOJc8uLq6uoNABaOC4cIXG8JU31R2B6ZJgAIIANxovmOnVuNyhEme6jeo0+9m2wAaABUYMIwMwaGgIEJwbxIddiWgAIwADiE-W+mhcbgA4pZssxKN1kkJEASztCiWA8LgEBsRKM9Gwcm5WQh2YgpnyAhAvKyIFgKMBVARwnAGtpkARMhAIMwekJ1QQ8Z4tWBdQAKNA6gh8mDTADmAEoCABeAB8BCBEDtpqwbPGiF1jIgzMl0oQqiMqlUCAAHoMCM8hHyqMGOdVxvtDp6CLF4ta2FAOUw8ZmpSGc3sDnFrUHlVaq9HZXh5YrOCr3WqhBq0xngEHq9mwmwIJaENowubJ0kTB6CE2VabB4QANqxOuHQpG7cVB0IPC7PNxPdxd4AXQLDWarQAPAAJVFVCYXg8EAA+BD2GS91p-swrprniQ54Iu9TJOimLYhWQjWiBJr9qmYGENifIAuUaBAggEyTjilZLjoeiGI2WaIKclhwIR6bfj+G5XmEG48BheBYUwuEAn0TEECxuhgHAPBXq6pxsNM2jWqaGrWtRtHut6BByfBpzaPOB5DDOppIQOaEELQ5BTDMBb6AaRrCLJ5DsdhuH4TR8EAPyqep7yaUhpoSEQRDSSQqK0GgFABWQBBeLguS+NAQjaAQ-IELkCAIFgmpJGa-Z0G4yRGuU06FrmBCiHEPkhbQ4wxSVSkBYQEBsFBeiBWl5XblA+UUFAWDWGM2g+RIpoNMUpTlNaiGej6fYahqaA1SW87JNFABkc25Tu577u8vbSeNTX1hgH7vGJM2IRtGpbbuq2jKcyTsXgeBOpkazWjwHWkA8mVlJ1PBhDwPA6eNcbIeN+VTQAhBOU7aOtKEAydK2XqMTndPEXHXbd92PaCL0IFl70-Rqf248x0M7Wdh6g9qInxih676cI2hGhaKVLg0aQZNkeRDQpo0bXOlrkaOlENDApx2j9RjMdzSRhCO0anALQvaOT-2oemhBGbkKQvHi7y3skNTFhtw3ekdBASBIBBSsw0x4A82gBaQmRGg8byJlbFtvAQfB0oJTpYFb5uWxIQgQC9BlqV4RsTjAPh+EEbAlmW2GVqr6twu8snTLkSoqmE3RsLqniR5E04x8+r4TACHK0Lgu2jF6EMAwDk0ENaQNsRxOF4QRjmnPIdDut0eCwP2QEU-X4394PTdGwDD4jGMkzp1P9fAOcVHp5nYB46P9eW+8HrKCA06iPvi8A-IGIYGAHrAGNW+j3auqtzZHf2biQhOWpWoaVpkO3+NPe0LqagdBGAsDAKcfkzQnRCGdJZTCT87K0Scv-V0IQT6-U3r-AgOcr45wwVvCQXpF441+igjaG5H6cWfrRJi2kR5YOPBPKSP8Hynm2kbcgV8KHtwQfBAgDkCBJw1vCQgupHo4R4HgjUT0HgcP3qQY+P8pG23ttOK+QMDJgFVpI1K9NL7AAMlFOmM1tEjiBGgBAXgr4Pj+HoYEoIYBxQQGIK+Usawy3+HLEwYA0AAC8EBX1JCYAWLiKIIHcXoEwBDtHYOAJdVhp1YYQSNmMZ2IS+ZhNljbMAWBsRiAAHKkFiNo-khomBYHXlfReGjVZ8PqFFYI0wcq6mAGmOeeBdR4Aeq47MpxA5W2mDI62LlRjfUltiRAB5RFGjYH0bReCCG0PxEhEARggA).
[I tried removing the `useMemo` to leave it to the compiler how to memoize it, and the bug was still there](https://playground.react.dev/#N4Igzg9grgTgxgUxALhASwLYAcIwC4AEAVAQIZgEAKpeAFgKp5oA2FAZjBBgQOQACYUgDs0eAJ4B6KE2YSsNWmB4AdEdlyFguACYIYAITEBfAhy69mEbeVoq1OfAWDisCAgDFcAd1Ixd2gCUENgAaU29fQOCwjAQMCDCoMAQAYVJmZgAjUjgAa0TkgFE2NgQ4PAKEAFk4hIIkhABlPBoEEzNuHhgEHLw7TAdNAEkhPARRgBk0IVz2zk7BEXEJTmk9O1UBjScaoSghsYw58x4AOlOJc8uLq6uoNABaOC4cIXG8JU31R2B6ZJgAIIANxovmOnVuNyhEme6jeo0+9m2wAaABUYMIwMwaGgIEJwbxIddiWgAIwADiE-W+mhcbgA4pZssxKN1kkJEASztCiWA8LgEBsRKM9Gwcm5WQh2YgpnyAhAvKyIFgKMBVARwnAGtpkARMhAIMwekJ1QQ8Z4tWBdQAKNA6gh8mDTADmAEoCABeAB8BCBEDtpqwbPGiF1jIgzMl0oQqiMqlUCAAHoMCM8hHyqMGOdVxvtDp6CLF4ta2FAOUw8ZmpSGc3sDnFrUHlVaq9HZXh5YrOCr3WqhBq0xngEHq9mwmwIJaENowubJ0kTB6CE2VabB4QANqxOuHQpG7cVB0IPC7PNxPdxd4AXQLDWarQAPAAJVFVCYXg8EAA+BD2GS91p-swrprniQ54Iu9TJOimLYhWQjWiBJr9qmYGENifIAuUaBAggEyTjilZLjoeiGI2WaIKclhwIR6bfj+G5XmEG48BheBYUwuEAn0TEECxuhgHAPBXq6pxsNM2jWqaGrWtRtHut6BByfBpzaPOB5DDOppIQOaEELQ5BTDMBb6AaRrCLJ5DsdhuH4TR8EAPyqep7yaUhpoSEQRDSSQqK0GgFABWQBBeLguS+NAQjaAQ-IELkCAIFgmpJGa-Z0G4yRGuU06FrmBCiHEPkhbQ4wxSVSkBYQEBsFBeiBWl5XblA+UUFAWDWGM2g+RIpoNMUpTlNaiGej6fYahqaA1SW87JNFABkc25Tu577u8vbSeNTX1hgH7vGJM2IRtGpbbuq2jKcyTsXgeBOpkazWjwHWkA8mVlJ1PBhDwPA6eNcbIeN+VTQAhBOU7aOtKEAydK2XqMTndPEXHXbd92PaCL0IFl70-Rqf248x0M7Wdh6g9qInxih676cI2hGhaKVLg0aQZNkeRDQpo0bXOlrkaOlENDApx2j9RjMdzSRhCO0anALQvaOT-2oemhBGbkKQvHi7wFhOMA+H4QRsCWZbYZWqvq3C7yydMuRKiqYTdGwuqeLrkTTgbz6vhMAIcrQuC7aMXoQwDk0ENaQNsRxOF4QRjmnPIdDut0eCwP2QEUwDScp6HR0EA+IxjJM1s5xqwDnFR1u22AeMA+N0wF3gHrKCA06iE3xfLr4pAYGAHrAGNNfB-aEc2dH9m4kITlqVqGlaZDA8d3QurUHQjAsGApz8s0TpCM6lmYSPdm0U58e0K6ITt0Y1cDw7vcO1f40SF6G0i+5KGZzA-ZSXPD6nttOfkL3YenFR60QIA5AgZsNbwkILqR6OEeD3w1E9B4ACm6kDbnPJBAVSCZCNNoXuQMDJgFVog1K9Me7AAMlFOmM1SEjiBGgBAXhe4Pj+HoYEoIYBxQQGIXuUsawy3+HLEwYA0AAC8EC91JCYAWfCKIIEEXoEwj9SE32AJdX+p1YYQRzmMRMDdhzyMUYLbQAUsDYjEAAOS7m0XRZkmBYErr3duRDVZgPqFFYI0wcq6mAGmeuuo8APX4dmU4QgIB4AeNMFBDwp5QAPN9SW2JEAHlgUaNgfRSH30ftpWMSEQBGCAA).

[Turns out the problem is that defining components inline like this isn't a supported pattern.](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unstable-nested-components.md)
By refactoring `<MenuItem>` to use a stable `as` prop to `IntentLink`, and give it props and live it to `styled-components` to forward props, [it works with the compiler as well.](https://playground.react.dev/#N4Igzg9grgTgxgUxALhASwLYAcIwC4AEAVAQIZgEAKpeAFgKp5oA2FAZjBBgQOQACYUgDs0eAJ4B6KE2YSsNWmB4AdEdlyFgBXABMEMAEJiCAXwIcuvZhB3laKtTnwEtGBBggAaAlDAIAwqTMzABGpHAA1t6+CACibGwIcHjRfgDKeDQIpuac3DwwCOF4DphOmgQAkkJ4CDUAMmhCETkW+YIi4hKc0voOqmUaLgQAsnVQlbXcZm28AHRzEgtLi8vLUGgAtHBcOEJ1eEoD6s5a9H4wAIIAbjSkMK1586svKxI76vs1R45DWjEAFRgwjAzBoaAgQkelh4a1eyzQAEYABxCUonCriLDZADi1jCzEohT8QkQ0PycLeyzAeFwCH6Ihq+jY4WyRIQJMQjRpACUIAB3IkQLAUYCqAjmCBwGI6ZAEEIQCDMIpCcXaIQAMSlvjlAAo0LKCDSYE0AOYASgIAF4AHwEa4QA1q6xwcGQsByvEQAnszkIADaPBdbqESgAumqYjBPfigr66ohA1GeBGhCZVKoEAAPcoEHahwjx0kIMZCCZTa0ENweXVsKCkpiQqjEhMl8aTdy6rCcEVyotctC8gVCkWWsVQvPuirBxuh1L6bxsbV+HTeSFa6UUMxWgjd4VgNX5mkEf1uMsdjCxZVnlJGhB4Uvl9xX9wHMOVmIZLIEAA8BAAEgCIz1C+N4EAAPgQZbBAQdq6tBzDmoeU7DIQ24+H4QIgmCs66khqoThIRBEGqxAEACtCDgQVGkAQ-K4BE9zQEIOgELSBARAgCBYJKm7qmxtDZH4yrJAgrFnlA1FTKRJD8oJUJ0NkzCDoQEBsBh+gUE0AnZBJ1EUFAWC2LUOgyRIkZ+PEiTJLqeHWna44ShKaDqbWy5iQQABknlVu2UygQcY6kU5EkXgFNRzEum54cFEqhf514HHMfh4JceB4CaIS9LqPDGaQmzCUkJk8N4PA8PhTmmBmE7Oa5ACEUUykFNUhX5z6JTUAD8cyFB41wIGlGVoFltQ5XlBUICJxUVU56YTiY3inm1l4dbejUrmG+HIQWBC0MIOjKhuvgfn4gTBGEkS2ZatouMF67LrqUZzAaM0LSe92bvOMDPTom3VRKR6EGCNKXMkaD9fUUohpWuj6EYuozhCoYQZB-photQbkKlYP9WlKYY3oYBwCm5qRU0Oi6sFCNQ7O112ojkJzDo2o3pUq5qjNgO7eQjTNJWBiKsqwgI1joNMBDNNI91zPSqzOhbROLkECLIM4wgkOurTt0tYUeCwFClMtRKP6PhesVOeQVrANUtQNE0ERzZVlUSBIBB8IcmzZtiySezAnAPJsBAAAZNLbeBB-p9pBAadHyRhZrBzbBy8xEQfm85TI1FayggGJog5+nu73KQGBgFbjlO5VBpysD2Pi+rkuMzLUBy54hcSvIdB9gojAsGAcy0hkJpCKaKt1+DDea0jcyd7Q5pt0bs2O5XBAuwQaR7YUrF7iKhfjZbOekAXi8Sh9vhW3tLGHcuy+V92CDXGgCD8lbP7nPoNx3A8YBoAAXggVtERmCjFbKMZgJA2lvk7QobArYpVNglV8NQoGVVqFmPAoCLhM0HFgMEYgAByJcEAoNXjaYKr1-oEF1vrZWwUTbLXNgfEAOhwbH0riwwQIRlQ6CtnVPaYAU4kNpEqJgWARxlwrpXfhKcCCdR8CxBAbAmgeTlFofMYc5R4ByvfP0cwhAQDwJsJomx8rNxvOVbwuDWQ3jlEGRRJQqqLxIWvDe9wPI7wPIvfeZdD5sKdmfCRl8DoICOmAEh99H7P1fu-K4txMjfz-gA4AQCNIwEwfocBkDzYwLgfeBB7UkF4CEdmDBwAnocKsQQohUCIEc1UCYfCIATBAA)